service/collector/collector.go

```diff
@@ -75,9 +77,21 @@ func New(config Config) (*Collector, error) {
 
 func (c *Collector) Boot(ctx context.Context) {
 	c.bootOnce.Do(func() {
+		{
+			c.logger.LogCtx(ctx, "level", "debug", "message", "registering collector")
+
+			err := prometheus.Register(c)
+			if IsAlreadyRegisteredError(err) {
+				c.logger.LogCtx(ctx, "level", "debug", "message", "collector already registered")
+			} else if err != nil {
+				c.logger.Log("level", "error", "message", "registering collector failed", "stack", fmt.Sprintf("%#v", err))
+			} else {
+				c.logger.LogCtx(ctx, "level", "debug", "message", "registered collector")
+			}
+		}
```

Comment:
I moved the registration out of the service because we have a `Boot` interface. There will be more work being done towards getting the interface straight and cleaning this up though.

Reply:
Yeah, we can probably abstract this nicely.